eigen/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+idf_component_register(
+                       # We need the dummy source file so that the component
+                       # library is not an interface library. This allows to
+                       # get the list of include directories from other components
+                       # via INCLUDE_DIRECTORIES property later on.
+                       SRCS dummy.c
+                       # Eigen librarires depend on some of the IDF libraries,
+                       # add them here:
+                       PRIV_REQUIRES vfs lwip pthread newlib)
+
+# Determine compilation flags used for building Eigen
+# Flags inherited from IDF build system and other IDF components:
+set(idf_include_directories $<TARGET_PROPERTY:idf::eigen,INCLUDE_DIRECTORIES>)
+set(includes "-I$<JOIN:${idf_include_directories}, -I>")
+# IDF is built with -D_GNU_SOURCE, but we don't actually implement all _GNU_SOURCE
+# features. Defining _GNU_SOURCE causes Eigen to try to use them, which fails.
+# Use _DEFAULT_SOURCE instead, it provides enough features to build Eigen.
+set(extra_defines -D_DEFAULT_SOURCE)
+# Final flags for C and C++:
+set(c_flags "${includes} ${extra_defines} -D_DEFAULT_SOURCE=1")
+set(cxx_flags "${includes} ${extra_defines} -D_DEFAULT_SOURCE=1")
+
+set(assert_flags "${assert_flags} -DDEBUG -D_DEBUG -ffunction-sections -fdata-sections ")
+
+if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)
+    set(opt_flags "-Og -ggdb -mlongcalls ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug
+                 -DCMAKE_C_FLAGS_DEBUG=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_DEBUG=${opt_flags})
+elseif(CONFIG_COMPILER_OPTIMIZATION_SIZE)
+    # Normally we should use MinSizeRel build type for this,
+    # however Eigen CMake files only handle Debug and Release
+    # (see e.g. Eigen/cmake/EigenCompilerOptions.cmake).
```

Comment:
There is no EigenCompilerOptions.cmake in https://gitlab.com/libeigen/eigen/-/tree/master/cmake...

It looks like this part has been copy-pasted from OpenCV CMakeLists.txt file.